### PR TITLE
MDS-3788 Show EPIC Authorizations on mines that have them in EPIC

### PR DESCRIPTION
**Pull request by JustinMacaulay**
The snap shot tests are not done yet, just wanted what I have no up so ppl could start reviewing it.

Adds 2 new services for NRPTI and EPIC
A new resource for grabbing the EPIC information
A new section under permits and approvals for external authorizations.

### CODE REVIEW

File: services/core-api/app/api/mines/external_authorizations/resources/epic_resource.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from flask.globals import current_app
```
Comment (mmumby) [inline services/core-api/app/api/mines/external_authorizations/resources/epic_resource.py]: remove all unused imports

File: services/core-api/app/api/services/nrpti_service.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import requests
+import json
+
+from flask import current_app
+
```
Comment (mmumby) [inline services/core-api/app/api/services/nrpti_service.py]: remove

File: services/core-web/common/actions/mineActions.js [hunk 3]
```diff
@@ -40,3 +40,8 @@ export const storeMineComments = (payload) => ({
   type: ActionTypes.STORE_MINE_COMMENTS,
   payload,
 });
+
+export const storeEpicInfo = (payload) => ({
+  type: ActionTypes.STORE_MINE_EPIC_INFO,
```
Comment (mmumby) [inline services/core-web/common/actions/mineActions.js:45]: can you create an `ExternamAuthorizationsActions.js` for this function to be consistent with the other redux

File: services/core-web/common/reducers/externalAuthorizationReducer.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import * as actionTypes from "../constants/actionTypes";
+import { EXTERNAL_AUTHS } from "../constants/reducerTypes";
+
+/**
+ * @file mineReducer.js
+ * all data associated with new mine/existing mine records is handled witnin this reducer.
```
Comment (mmumby) [inline services/core-web/common/reducers/externalAuthorizationReducer.js]: update comment

File: services/core-web/common/selectors/externalAuthorizationSelectors.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import * as externalAuthReducer from "../reducers/externalAuthorizationReducer";
+/* eslint-disable import/prefer-default-export */
+export const getMineEpicInfo = (state) => externalAuthReducer.getMineEpicInfo(state);
```
Comment (mmumby) [inline services/core-web/common/selectors/externalAuthorizationSelectors.js]: Can you update to use the newer export syntax? 

```
export const {getMineEpicInfo} = externalAuthReducer
```

File: services/core-web/src/components/mine/ExternalAuthorizations/ExternalAuthorizationsTable.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+import PropTypes from "prop-types";
+import * as Strings from "@common/constants/strings";
+import CustomPropTypes from "@/customPropTypes";
+import CoreTable from "@/components/common/CoreTable";
+import LoadingWrapper from "@/components/common/wrappers/LoadingWrapper";
+import NullScreen from "@/components/common/NullScreen";
+
+/**
+ * @class MineExplosivesPermitTable - list of mine explosives storage and use permits
+ */
```
Comment (mmumby) [inline services/core-web/src/components/mine/ExternalAuthorizations/ExternalAuthorizationsTable.js]: update comment

File: services/core-web/src/customPropTypes/externalAuthorizations.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { PropTypes } from "prop-types";
```
Comment (mmumby) [inline services/core-web/src/customPropTypes/externalAuthorizations.js:1]: niccce!

**Review by mmumby — COMMENTED** [review]
Great work, Justin!! Excited to get this into Prod. 
Few minor comments and there are two .py files that are empty that I can't comment on, can those be deleted?

File: services/core-web/src/customPropTypes/externalAuthorizations.js (hunk 7 above)
Comment (JustinMacaulay) [reply services/core-web/src/customPropTypes/externalAuthorizations.js:1]: not sure what this one is about

File: services/core-web/src/components/mine/ExternalAuthorizations/ExternalAuthorizationsTable.js (hunk 6 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/mine/ExternalAuthorizations/ExternalAuthorizationsTable.js]: Done

File: services/core-web/common/selectors/externalAuthorizationSelectors.js (hunk 5 above)
Comment (JustinMacaulay) [reply services/core-web/common/selectors/externalAuthorizationSelectors.js]: Done

File: services/core-web/common/reducers/externalAuthorizationReducer.js (hunk 4 above)
Comment (JustinMacaulay) [reply services/core-web/common/reducers/externalAuthorizationReducer.js]: Done

File: services/core-web/common/actions/mineActions.js (hunk 3 above)
Comment (JustinMacaulay) [reply services/core-web/common/actions/mineActions.js:45]: Done

File: services/core-api/app/api/services/nrpti_service.py (hunk 2 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/services/nrpti_service.py]: Done

File: services/core-api/app/api/mines/external_authorizations/resources/epic_resource.py (hunk 1 above)
Comment (JustinMacaulay) [reply services/core-api/app/api/mines/external_authorizations/resources/epic_resource.py]: Done

File: services/core-web/src/customPropTypes/externalAuthorizations.js (hunk 7 above)
Comment (mmumby) [reply services/core-web/src/customPropTypes/externalAuthorizations.js:1]: was just commenting on the custom proptype

**Review by mmumby — DISMISSED** [review]
🔥  well done!

**Review by mmumby — APPROVED** [review]

**Comment by jon-funk** [comment]
push

**Review by jon-funk — CHANGES_REQUESTED** [review]
Are the EPIC and NRPTI services covered under any backend tests?

**Review by jon-funk — APPROVED** [review]

**Pull request merged**